src/snowflake/snowpark/session.py

```diff
@@ -1623,8 +1623,14 @@ def query_history(self) -> QueryHistory:
         return query_listener
 
     def _table_exists(self, table_name: str):
-        tables = self._run_query(f"show tables like '{table_name}'")
-        return tables is not None and len(tables) > 0
+        try:
+            # DESC is used here because SHOW does not work for qualified table name
+            self._conn._cursor.describe(f"DESC TABLE {table_name}")
+            return True
+        except ProgrammingError:
+            # We could verify if this is a compile error and throw otherwise, but directly return False gives no
```

Comment:
Ideally we can check whether `does not exist or not authorized` is in error message then return false but it might be fragile and unnecessary.